### PR TITLE
Compound terms for HubMap

### PR DESCRIPTION
**Pull request by anitacaron**
Context: https://github.com/hubmapconsortium/ccf-ontology/issues/16

Use of the DOS-DP `cellPartOfAnatomicalEntity` to generate cells location specific.

### CODE REVIEW

**Comment by anitacaron** [comment]
@shawntanzk @bvarner-ebi, could you double-check the new compound terms, please?

**Comment by anitacaron** [comment]
It might have redundant compound terms. Could you point out, please?

**Comment by shawntanzk** [comment]
> def:
  text: "Any %s that is part of a(n) %s."
  vars:
    - cell
    - anatomical_entity
    
not sure if template changes are up for discussion, but following most of CL, I'd rather `A %s that is part of a(n) %s.`

**Comment by anitacaron** [comment]
I checked the QC, and in the UBERON terms, there's an obsolete term. I'll remove it.

**Comment by shawntanzk** [comment]
@anitacaron - could you help provide the cellPartOfAnatomicalEntity table but with labels please? would help the review a lot. PS do let me know if you want me to take a look at the QC error, I briefly looked and I'm guessing you'll get to it too :)

**Comment by anitacaron** [comment]
> could you help provide the cellPartOfAnatomicalEntity table but with labels please

The idea of the pattern is to generate labels. You can check the `definitions.owl` file. 😃

**Comment by shawntanzk** [comment]
> The idea of the pattern is to generate labels. You can check the definitions.owl file.

Yeah I know, was just hoping to review the template file know the pattern - little easier to find redundant compound terms

**Comment by bvarner-ebi** [comment]
Hi @anitacaron, is there an associated ticket or discussion for this PR? I'd like to understand the rationale for creating new cell types solely based on anatomical location and potentially no other distinguishing characteristics.

**Comment by shawntanzk** [comment]
btws so we dont duplicate effort, @bvarner-ebi I'm doing edits already yeah

**Comment by shawntanzk** [comment]
PS removed 1 term cause it was unnessacary

**Comment by anitacaron** [comment]
@bvarner-ebi, sorry, I updated the issue's description with the context.

**Review by bvarner-ebi — APPROVED** [review]
Reviewed the term labels in the definitions.owl file. I did not identify gross errors or duplication.

I haven't been involved in the discussions surrounding this particular issue, but it appears that this approach may set an unsustainable precedent. If the intention is to add new classes for every cell type in every anatomical structure where it is present, CL could become bloated and unbalanced quite quickly.
If appropriate, I would consider if the use case could be met by using post-coordinated classes where the user combines the already existing ontology classes into more complex concepts on their end in lieu of creating novel CL classes for each instance.

**Comment by shawntanzk** [comment]
I'm guessing the use case (attaching markers to cell types) would be difficult to handle with complex concepts: 
If you want a knowledge graph which attaches a set of markers to a cell type that is easily queryable, you can technically have a node `cell x markers` that links to the markers, algorithm/method, and cell y and anatomical location z. querying becomes quite difficult unless you have some proper tooling, which then makes adaptability to other use cases difficult. I do agree that we want to avoid blowing up the ontology, but I think this is a really hard case without the cell type included. Anyway that's my thoughts on it :)

**Comment by addiehl** [comment]
I am in favor of these compound terms, particularly if they are to be further specialized with anatomy-specific markers, and as long as there is a sanity check to ensure only real cell types are created (i.e, don't blindly fill in a matrix of cell type x anatomy without experimental evidence)

**Comment by bvarner-ebi** [comment]
Hi, @anitacaron, can you clarify the action here? Can this be merged / closed?

**Comment by dosumis** [comment]
@anitacaron  - looks like we just need to add the references.  Would be great if we could get this merged before a release.

**Comment by anitacaron** [comment]
Yes, I'm working on this now.

**Comment by anitacaron** [comment]
I removed one row because the reference is ISBN-13:978-1496398949 (page 641). Is this a valid reference?

I didn't add the references for some rows because, after curation in this PR, the new cells are not used in the ASCT+B tables. Should we recommend changing these cells in the ASCT+B tables?

File: src/patterns/data/default/cellPartOfAnatomicalEntity.tsv
Comment (bvarner-ebi) [inline src/patterns/data/default/cellPartOfAnatomicalEntity.tsv:23]: Values for "defined_class_label" are missing for all rows.
Values for "pubs" missing for several rows.

Comment (anitacaron) [reply src/patterns/data/default/cellPartOfAnatomicalEntity.tsv:23]: I commented about the missing pubs here https://github.com/obophenotype/cell-ontology/pull/1722#issuecomment-1374109814 

The labels will be filled up with the DOS-DP. You need to check the `definions.owl` file to have a complete view.

**Comment by anitacaron** [comment]
@shawntanzk, can you explain the changes you made in the tsv file, please?

**Comment by shawntanzk** [comment]
> Curate away over compounded

I'll try but this was a long time ago so....

1) remove CL:6000019 which would be podocyte of the glomerular visceral epithelium (which is also using an obsolete uberon term. podocyte has the following equiv
```
'epithelial cell'
 and ('part of' some 'glomerular visceral epithelium')
 ```
 making the term redundant. 
 2) remove CL:6000025 which would be extravillous trophoblast of trophoblast - which again redundant

**Comment by shawntanzk** [comment]
#gogoeditdiff SHUT UP IM DOING THIS! lol

**Comment by anitacaron** [comment]
There are a couple of changes in this commit https://github.com/obophenotype/cell-ontology/pull/1722/commits/6b603c03b5d5a188ae664bd8da1040c3b182698f

**Comment by shawntanzk** [comment]
> There are a couple of changes in this commit [6b603c0](https://github.com/obophenotype/cell-ontology/commit/6b603c03b5d5a188ae664bd8da1040c3b182698f)

whoops:
Basically stripping our location stuff in CL terms (they should derive from the uberon term instead. 

CL:6000026 Change from paneth cell of colon to paneth cell, redundancy given uberon epithelium of crypt of Lieberkuhn used already
Same for CL:6000027 - (CL:0009011 - remove "of colon"), CL:6000027 (CL:0019032 - remove "of intestine"), CL:6000028, CL:6000029 (CL:1000279 - remove "of large intestine"), & CL:6000035 (CL:1000457 - remove "of visceral peritoneum"

**Comment by anitacaron** [comment]
> Should we recommend changing these cells in the ASCT+B tables?

**Comment by shawntanzk** [comment]
> Should we recommend changing these cells in the ASCT+B tables?

It's not technically wrong, and in fact correct for the purpose of curating CL terms, just too specific CL term for template. I think the aim would be to update the cell type with the newly created terms? so don't see a need to change, and changing would make the asct+b tables less precise

**Comment by shawntanzk** [comment]
#gogoeditdiff

**Comment by anitacaron** [comment]
#gogoeditdiff

File: src/patterns/definitions.owl
Comment (bvarner-ebi) [inline src/patterns/definitions.owl:65]: Did not find this specific cell type explicitly mentioned in the dbxref.

Comment (anitacaron) [reply src/patterns/definitions.owl:65]: @aleixpuigb, do you want to comment?

File: src/patterns/definitions.owl
Comment (bvarner-ebi) [inline src/patterns/definitions.owl:72]: Of note, the reference describes neuroendocrine cells in the "deep mucosal layer along the muscularis mucosa" in a subset of non-pathology specimens.

File: src/patterns/definitions.owl
Comment (aleixpuigb) [reply src/patterns/definitions.owl:65]: There is no mention to blood vessel endothelial cells in venules/arteries of the lymph node apart of the high endothelial cells in high endothelial venules (HEVs) in the literature. However there is reference of the artery and venules (non-HEVs), and according to literature, [all blood vessels have blood vessel endothelial cells](https://www.ncbi.nlm.nih.gov/books/NBK26848/#:~:text=Endothelial%20Cells%20Line%20All%20Blood%20Vessels&text=The%20amounts%20of%20connective%20tissue,endothelial%20lining%20is%20always%20present.):

"The amounts of connective tissue and smooth muscle in the vessel wall vary according to the vessel's diameter and function, but the endothelial lining is always present. "

I agree that the dbxref provided is not explicit about the existence of those cells. What would be your recommendation  for cells that are not mentioned in literature, but the existence is implicit, @bvarner-ebi?

Comment (dosumis) [reply src/patterns/definitions.owl:65]: I don't doubt that these cells exist.  I think reasonable inference is fine - blood vessels have endothelial cells & these blood vessels exist.  Would be strengthened by mention of markers for the endothelial layer - but I think OK without.  r just as Alex has used here. 

As these  terms come from ASCT+B tables, there are multiple expert authors.  We could try to get their ORCiDs.  Or we could just add a comment on origins.

File: src/patterns/definitions.owl
Comment (bvarner-ebi) [reply src/patterns/definitions.owl:72]: The same reference appears as three separate dbxrefs with different identifiers. My understanding is that PubMed Identifiers (PMID) are [preferred](https://oboacademy.github.io/obook/lesson/contributing-to-obo-ontologies/#adding-annotations) and that different dbxrefs pointing to the same resource is not standard.

Comment (dosumis) [reply src/patterns/definitions.owl:72]: Well spotted.  Problem is the source - ASCT+B @aleixpuigb  - we should check whether different identifiers are duplicates before adding.

Comment (anitacaron) [reply src/patterns/definitions.owl]: done

**Review by aleixpuigb — APPROVED** [review]

**Comment by anitacaron** [comment]
#gogoeditdiff

File: src/patterns/definitions.owl
Comment (bvarner-ebi) [inline src/patterns/definitions.owl:86]: PMID:10715339 seems to be behind a paywall and investigates pathological states. No mention of the cell type in abstract. 
PMID: [22320805](https://pubmed.ncbi.nlm.nih.gov/22320805) seems OK, although it describes gross radiographic studies and nothing at the cellular level.

Comment (bvarner-ebi) [reply src/patterns/definitions.owl:86]: Considering adding variant spellings of taenia (e.g., tenia) as synonyms.

File: src/patterns/definitions.owl
Comment (bvarner-ebi) [inline src/patterns/definitions.owl:95]: Consider label 'serous cell of bronchus submucosal gland' to align with text in the references and similar other labels in CL (e.g., 'serous cell of epithelium of bronchus').

File: src/patterns/definitions.owl
Comment (bvarner-ebi) [inline src/patterns/definitions.owl:102]: Is there a particular reason this is not labelled simply as 'endothelial cell of efferent lymphatic vessel'?

File: src/patterns/definitions.owl
Comment (bvarner-ebi) [inline src/patterns/definitions.owl:109]: Consider using the label 'brush cell of epithelium of lobar bronchus' or simply 'brush cell of lobar bronchus'.

File: src/patterns/definitions.owl
Comment (bvarner-ebi) [inline src/patterns/definitions.owl:116]: Consider simplifying label as 'venous endothelial cell of respiratory system', which is easier to conceptualise and aligns with language used in the reference: "venous ECs".

File: src/patterns/definitions.owl
Comment (bvarner-ebi) [inline src/patterns/definitions.owl:135]: The dbxref is behind a paywall and there is no mention of this specific cell type in the abstract.

File: src/patterns/definitions.owl
Comment (bvarner-ebi) [inline src/patterns/definitions.owl:142]: The dbxref is behind a paywall and there is no mention of this specific cell type in the abstract. 
"LM: longitudinal muscle" is noted in the abbreviations list.

**Review by bvarner-ebi — APPROVED** [review]
Please see inline comments for your consideration before merging.

Please also note many of these dbxrefs focused on particular species and the cell types may not apply to all Metazoa; however, species restraints have not been added to these new terms.

**Comment by dosumis** [comment]
Re name changes

* compound names that look redundant are a sign that the genus term is too specific.  We already fixed some of these, but clearly  not enough.
* Aligning with names in the literature via label or synonym is good if we have time and resources.  It can be achieved by editing the TSV.

File: src/patterns/definitions.owl
Comment (dosumis) [reply src/patterns/definitions.owl:116]: Having looked at this example, we agree that over-riding the automatically generated name with your suggestion would be best.  While analysing the ontology, we notice that the logical definition of blood vessel endothelial cell should be simplified from "[squamous epithelial cell](https://www.ebi.ac.uk/ols/ontologies/cl/terms?iri=http%3A%2F%2Fpurl.obolibrary.org%2Fobo%2FCL_0000076) and [part of](https://www.ebi.ac.uk/ols/ontologies/cl/properties?iri=http%3A%2F%2Fpurl.obolibrary.org%2Fobo%2FBFO_0000050) some [blood vessel endothelium](https://www.ebi.ac.uk/ols/ontologies/cl/terms?iri=http%3A%2F%2Fpurl.obolibrary.org%2Fobo%2FUBERON_0004638)" to just use 'endothelial cell' as a genus.

File: src/patterns/definitions.owl
Comment (dosumis) [reply src/patterns/definitions.owl:109]: In this case, we can automatically generate the label 'brush cell of epithelium of lobar bronchus'  by specifying 'brush cell' + 'epithelium of lobar bronchus' in the table.  From inspecting the Equivalent Class axioms in this branch of CL it is clear that some normalization and correction is needed.  All should have brush cell as the genus & have some appropriately specific Uberon term in the differentia.

**Comment by anitacaron** [comment]
#gogoeditdiff

**Review by dosumis — APPROVED** [review]
Looks good to me!  Happy to merge.

**Comment by dosumis** [comment]
DO NOT MERGE YET! @anitacaron will  first merge definitions.owl into editors file and move the TSV to an archive folder under src/patterns/data with a README in the folder making clear nature of archived fields.

**Comment by anitacaron** [comment]
#gogoeditdiff

File: src/patterns/definitions.owl
Comment (anitacaron) [reply src/patterns/definitions.owl]: done

**Comment by anitacaron** [comment]
Ready to merge. I double-checked the diff after merging the terms in `definitions.owl` into the `cl-edit.owl` file and confirmed nothing had changed.

**Pull request merged**
